### PR TITLE
Add support for devedition checks.

### LINKED ISSUE #143: Handle devedition as a product on its own with specific checks
For devedition we should have different checks
- [ ] Archive and Partner repacks https://archive.mozilla.org/pub/devedition/releases/58.0b1/
- [ ] Balrog should check the devedition rule
- [ ] Buildhub should check for the devedition product
- [ ] Download links should check for the devedition channel rather than Firefox beta one

### PR DESCRIPTION
**Pull request by Natim**
Fixes #143

### CODE REVIEW

**Comment by leplatrem** [comment]
@Natim what's the status of this? Is it ready to review?

**Comment by Natim** [comment]
@leplatrem This is good for review.

File: pollbot/tasks/balrog.py [hunk 1]
```diff
@@ -78,6 +78,8 @@
 
     elif channel is Channel.BETA:
         url = 'https://aus-api.mozilla.org/api/v1/rules/firefox-beta'
+        if product == 'devedition':
+            url = 'https://aus-api.mozilla.org/api/v1/rules/devedition'
```
Comment (leplatrem) [inline pollbot/tasks/balrog.py]: nit?
```py
path = 'devedition' if product == 'devedition' else 'firefox-beta'
url = f'https:// .... {path}'
```

File: pollbot/__init__.py [hunk 2]
```diff
@@ -4,4 +4,4 @@
 __version__ = pkg_resources.get_distribution(__package__).version
 
 HTTP_API_VERSION = '1.2'
-PRODUCTS = ['firefox']
+PRODUCTS = ['firefox', 'devedition']
```
Comment (leplatrem) [inline pollbot/__init__.py]: nit: could be a tuple :)

File: tests/test_views.py [hunk 3]
```diff
@@ -533,6 +601,16 @@ def cli(loop, test_client):
     assert body['link'] == "https://www.mozilla.org/en-US/firefox/all/"
 
 
+async def test_devedition_bedrock_download_links(cli):
+    resp = await check_response(cli, "/v1/devedition/58.0b15/bedrock/download-links")
+    body = await resp.json()
```
Comment (leplatrem) [inline tests/test_views.py]: Will this always work in the future?

File: tests/test_views.py [hunk 4]
```diff
@@ -259,6 +273,33 @@ def cli(loop, test_client):
...
 
 
+async def test_get_checks_for_devedition(cli):
+    await check_response(cli, "/v1/devedition/56.0b6", body={
+        "product": "devedition",
+        "version": "56.0b6",
+        "channel": "beta",
+        "checks": [
+            {"url": "http://localhost/v1/devedition/56.0b6/archive", "title": "Archive Release",
+             "actionable": True},
```
Comment (leplatrem) [inline tests/test_views.py]: meganit: is it me or your indentation is weird?

```py
"checks": [{
        "url": "http://localhost/v1/devedition/56.0b6/archive",
        "title": "Archive Release",
        "actionable": True
    },
]
```

File: tests/test_views.py (hunk 3 above)
Comment (Natim) [reply tests/test_views.py]: Good catch !

**Pull request merged**